ui/src/createAzureClient.ts

```diff
@@ -43,6 +46,63 @@ export async function createGenericClient(clientInfo?: ServiceClientCredentials
 
     return new gsc.GenericServiceClient(credentials, <types.IMinimumServiceClientOptions>{
         baseUri,
-        userAgent: appendExtensionUserAgent
+        userAgent: appendExtensionUserAgent,
+        requestPolicyFactories: addAzExtFactories
     });
 }
+
+function addAzExtFactories(defaultFactories: RequestPolicyFactory[]): RequestPolicyFactory[] {
+    // NOTE: Factories at the end of the array are executed first, and we want these to happen before the deserialization factory
+    defaultFactories.push(
+        {
+            create: (nextPolicy, options): RequestPolicy => new RemoveBOMPolicy(nextPolicy, options)
+        },
+        {
+            create: (nextPolicy, options): RequestPolicy => new MissingContentTypePolicy(nextPolicy, options)
+        }
+    );
+    return defaultFactories;
+}
+
+const contentTypeName: string = 'Content-Type';
+
+/**
+ * Removes the BOM character if it exists in bodyAsText for a json response, to prevent a parse error
+ */
+class RemoveBOMPolicy extends BaseRequestPolicy {
+    constructor(nextPolicy: RequestPolicy, requestPolicyOptions: RequestPolicyOptions) {
+        super(nextPolicy, requestPolicyOptions);
+    }
+
+    public async sendRequest(request: WebResourceLike): Promise<HttpOperationResponse> {
+        const response: HttpOperationResponse = await this._nextPolicy.sendRequest(request);
+        const contentType: string | undefined = response.headers.get(contentTypeName);
+        if (contentType && /json/i.test(contentType) && response.bodyAsText) {
+            response.bodyAsText = removeBom(response.bodyAsText);
+        }
+        return response;
+    }
+}
+
+/**
+ * The Azure SDK will assume "JSON" if no content-type is specified, which can cause false-positive parse errors.
+ * This will be a little smarter and try to detect if it's json or generic data
+ */
+class MissingContentTypePolicy extends BaseRequestPolicy {
+    constructor(nextPolicy: RequestPolicy, requestPolicyOptions: RequestPolicyOptions) {
+        super(nextPolicy, requestPolicyOptions);
+    }
+
+    public async sendRequest(request: WebResourceLike): Promise<HttpOperationResponse> {
+        const response: HttpOperationResponse = await this._nextPolicy.sendRequest(request);
+        if (!response.headers.get(contentTypeName) && response.bodyAsText) {
+            try {
+                parseJson(response.bodyAsText);
+                response.headers.set(contentTypeName, 'application/json');
+            } catch {
+                response.headers.set(contentTypeName, 'application/octet-stream');
```

Comment:
Is it worth checking to see if there's any binary characters before setting this as the content type rather than just using `text/plain?`  How does the sdk handle octet-stream responses?

Reply:
These are why I chose octet-stream:

https://stackoverflow.com/questions/12539058/is-there-a-default-mime-type/12560996
https://stackoverflow.com/questions/1176022/unknown-file-type-mime

As far as I can tell, the sdk handles text/plain and application/octet-stream the same.